modules/gapi/src/api/gmat.cpp

```diff
@@ -36,6 +36,39 @@ const cv::GOrigin& cv::GMat::priv() const
     return *m_priv;
 }
 
+int cv::checkVector(const cv::GMatDesc& in, const int n, const int ddepth)
+{
+    GAPI_Assert(ddepth == in.depth || ddepth < 0);
+    const int chan = in.chan;
+    const cv::Size size = in.size;
+    int quantity = -1;
+    if (n <= 0)
+    {
+        if (size.width == 1 || chan == 1)
+        {
+            quantity = size.height;
+        }
+        else if (size.height == 1)
+        {
+            quantity = size.width;
+        }
+        // else input Mat can't be described as vector of points
+    }
+    else // n is given
+    {
+        if (chan == n && size.height == 1)
+        {
+            quantity = size.width;
```

Comment:
quantity is equal to size.width/height. Will it be difficult to understand if you combine these conditions?

Reply:
@mpashchenkov Do you mean, the cases with `n < 0` and `n > 0`?

Reply:
```
bool per = n > 0 ? chan == n : true;

if (size.height == 1 && per)
{
    quantity = size.width;
}
else if (size.width == 1 && per || (chan == 1 && (n > 0 ? size.width == n : true))) 
{
    quantity = size.height;
}
```

Something like this +-

Reply:
instead 45 - 68

Reply:
Now, with the latest changes, it is impossible to combine

Reply:
```
bool per = n > 0 ? chan == n : true;
if ((size.width == 1) && per)
{
     quantity = size.height;
     dimensionality = chan;
}
else if ((chan == 1) && n > 0 ? size.width == n : true)
{
     quantity = size.height;
     dimensionality = n > 0 ? n : size.width;
}
else if ((size.height == 1) && per)
{
     quantity = size.width;
     dimensionality = chan;
}
```
+-

Reply:
If it works that it can be simplified. MB

Reply:
Yes, you are right. It is debatable whether the function is clearer now, but it isn't difficult to figure out what it does, I hope.
My implementation slightly differs from yours, hope you'll accept mine